### DISCUSSION TITLE
Support for multiple versions of the same component

### DISCUSSION THREAD

**Discussion by MuraliNunna** [WeblateOrg/weblate#12300] [Q&A]
Hello,

I would like to configure my component, such that for every new feature that the developer works on, he can push the new updates to the files, and translators can work on them and once done, i would like weblate to raise a PR for that branch.

Currently, i see that this flow works for a single branch mapped to the component. I couldn't find any documentation regarding support for multiple branches for a component. A workaround can be to duplicate each component for each branch, which seems like a lot of work.

Isn't this a normal workflow for continuous translations? Or am i missing anything and overcomplicating the overflow? I'm open to suggestions on recommended way of continuous translations by weblate.

For ex.

`component1` repo in github -> `main` branch has the base translation files -> mapped to `component1` with branch as `main` and push_branch as `weblate_main`

Now, when a new feature comes in, developers would want to add new strings to the base language file in a new branch `feature1` -> how do we setup translation management for this branch in weblate WITHOUT creating a duplicate component like earlier?

**Comment by nijel** [comment]
There is no automation to translate multiple branches. I know some users are automating this via API, but generally workflows for this differ and nobody yet come with an idea how to integrate this reasonably into Weblate.

You can easily create an additional component for a different branch of an existing component when creating a new one (I've recently described this in https://github.com/WeblateOrg/weblate/issues/12487).

**Comment by MortenKaae** [comment]
A bit late to the party here, but we just switched away from Transifex to Weblate due to the pricing, and I also found that Weblate really lacks in terms of multi-branch support compared to Transifex.

Transifex has very nice handling of this, where they automatically import new translations when a pull request is opened in e.g. GitHub from any branch, from which you can translate them in their interface. Once translated, it will make a commit in the PR with the changes.

This is a great flow for working on many branches within a repository, and I would love to hear if anyone has found a way to achieve a similar flow in Weblate?

**Reply by nijel** [reply]
Can you please describe that in more detail for our issue tracker: https://github.com/WeblateOrg/weblate/issues/new?template=feature_request.yml

PS: Currently it would be blocked by https://github.com/WeblateOrg/weblate/issues/1688.
